tensorflow_addons/callbacks/tqdm_progress_bar.py

```diff
@@ -115,9 +121,81 @@ def on_train_begin(self, logs=None):
         self.total_steps = self.params["steps"]
 
     def on_train_end(self, logs={}):
+        self.is_training = False
         if self.show_overall_progress:
             self.overall_progress_tqdm.close()
 
+    def on_test_begin(self, logs={}):
+        if self.is_training is False:
+            self.test_num_epochs = self.params["epochs"]
+            # set counting mode
+            self.mode = "steps"
+            self.test_total_steps = self.params["steps"]
+            if self.show_epoch_progress:
+                self.test_epoch_progress_tqdm = self.tqdm(
+                    total=self.test_total_steps,
+                    desc="Evaluating",
+                    bar_format=self.epoch_bar_format,
+                    leave=self.leave_epoch_progress,
+                    dynamic_ncols=True,
+                    unit=self.mode,
+                )
+            self.test_num_samples_seen = 0
+            self.test_steps_to_update = 0
+            self.test_steps_so_far = 0
+            self.test_logs = defaultdict(float)
+
+    def on_test_batch_end(self, batch, logs={}):
+        if self.is_training is False:
+            if self.mode == "samples":
+                batch_size = logs["size"]
+            else:
+                batch_size = 1
+
+            self.test_num_samples_seen += batch_size
+            self.test_steps_to_update += 1
+            self.test_steps_so_far += 1
+
+            if self.test_steps_so_far < self.test_total_steps:
+
+                for metric, value in logs.items():
+                    self.test_logs[metric] += value * batch_size
+
+                now = time.time()
+                time_diff = now - self.test_last_update_time
+
+                if self.show_epoch_progress and time_diff >= self.update_interval:
+
+                    # update the epoch progress bar
+                    metrics = self.format_metrics(
+                        self.test_logs, self.test_num_samples_seen
+                    )
+                    self.test_epoch_progress_tqdm.desc = metrics
+                    self.test_epoch_progress_tqdm.update(self.test_steps_to_update)
+
+                    # reset steps to update
+                    self.test_steps_to_update = 0
+
+                    # update timestamp for last update
+                    self.test_last_update_time = now
+            else:
+                self.test_logs = logs
+
+    def on_test_end(self, logs={}):
+        if self.is_training is False and self.show_epoch_progress:
```

Comment:
Since how the `model.evaluate()` is just having 1 epoch, and because there will only be one progress bar for `evaluate()`, I think we can safely assume that the user would want to see it and thus we won't need to check if `self.show_epoch_progress` is set. This is my opinion.